nntrainer/src/layer.cpp

```diff
@@ -215,4 +216,20 @@ void Layer::ensureName() {
   }
 }
 
+void Layer::print(std::ostream &out) const {
+  /// @todo change type enum to actual string value
+  out << '<' << typeid(*this).name() << "Object at" << this << '>' << std::endl
+      << "Layer Type: " << type << std::endl
+      << "Activation type: " << activation_type << std::endl
+      << "bias_initiate_to_zero: " << bias_init_zero << std::endl
+      << "Weight Decay: " << static_cast<int>(weight_decay.type) << " | "
+      << weight_decay.lambda << std::endl
+      << "Weight Initiation Type: " << weight_ini_type << std::endl
+      << "Loss: " << loss << std::endl
+      << "Cost: " << cost << std::endl
+      << "last layer: " << last_layer << std::endl
+      << "Need flatten: " << flatten << std::endl
```

Comment:
As flatten is always added as a new layer, lets not print it here. Flatten can be printed here if print in flatten layer is disabled. Same applies for activation as well.

Reply:
before init() is called, flatten layer isn't attached. so IMO it is needed. Instead I clarified name to "flatten layer after"

Reply:
Then lets give this only if the layer is not initialized. Same for activation.
It will be awkward to give it out twice, as layer property and a new layer after the layer is initialized.

Reply:
yes